### DISCUSSION TITLE
Add cookie parameters like domain, path... to the request cookies in middleware.tsx

### DISCUSSION THREAD

**Discussion by sanskar-19** [vercel/next.js#62799] [Ideas]
### Goals

1. When using `request.cookies.set()` include basic cookie attributes other than just name and value.

### Background

The problem that I am facing is cookie sharing across domains and subdomains which I have managed to implement on the client side (NextJS 14 App Router) but in the middleware, I get options for NextResponse to set, whilst not for NextRequest. 

Since there is no provision to update tokens on the client from Page components so I am updating the auth tokens by validating on the middleware only.

If in case of invalid tokens, I am refreshing the same and likewise updating a few parameters. 

By default if I set a cookie on `abc.com` it sets the domain parameter as `abc.com` but to access on subdomains as well, I need the parameter as `.abc.com`.

### Proposal

Add the options parameter to the request cookies in the middleware file.

**Comment by Netail** [comment]
You can already, as seen [here](https://nextjs.org/docs/app/building-your-application/routing/middleware#using-cookies).

```ts
response.cookies.set({
  name: 'vercel',
  value: 'fast',
  path: '/',
})
```
